app/commands/exercise/representation/send_new_feedback_notifications.rb

```diff
@@ -0,0 +1,36 @@
+class Exercise::Representation::SendNewFeedbackNotifications
+  include Mandate
+
+  queue_as :notifications
+
+  initialize_with :representation
+
+  def call = iterations.each { |iteration| send_notification(iteration) }
+
+  private
+  def iterations
+    if representation.has_essential_feedback?
+      latest_active_iterations
+    elsif representation.has_actionable_feedback?
+      latest_recent_active_iterations
+    else
+      raise "Incorrect feedback type"
+    end
+  end
+
+  def submission_representations = representation.submission_representations
+  def latest_active_iterations = submission_representations.map(&:iteration).select(&:latest?)
```

Comment:
This feels like it's going to be hellish from a performance perspective. Imagine for where there are thousands of latest iterations.

Reply:
Yes, this was just a quick and dirty first pass

Reply:
I've now refactored this to only run one query and the result appears to be quite fast. It ran in production in 0.02 seconds when tested on my Elixir high-score representation (248 matches). The query cost is around 980.

The resulting query is:

```sql
SELECT 
    `iterations`.`id`
FROM
    `iterations`
        LEFT JOIN
    `iterations` AS `i` ON `i`.`solution_id` = `iterations`.`solution_id`
        AND `i`.`deleted_at` IS NULL
        AND `i`.`idx` > `iterations`.`idx`
WHERE
    `iterations`.`deleted_at` IS NULL
        AND (`i`.`id` IS NULL)
        AND `iterations`.`id` IN (SELECT 
            `iterations`.`id`
        FROM
            `submission_representations`
                INNER JOIN
            `submissions` ON `submissions`.`id` = `submission_representations`.`submission_id`
                INNER JOIN
            `iterations` ON `iterations`.`submission_id` = `submissions`.`id`
        WHERE
            `submission_representations`.`ast_digest` = '590c2e4d266ec9a7ec5967db46145391640dcbbf');
```